### PR TITLE
Some changes to exec (bit simplication and check vat.Line)

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

File: src/DssDirectDepositHub.sol [hunk 1]
```diff
@@ -298,39 +305,41 @@ contract DssDirectDepositHub {
...
+            uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 toUnwind;
+            if (Art > lineWad) {
+                toUnwind = Art - lineWad;
+            }
+            if (debt > Line) {
+                toUnwind = _max(toUnwind, _divup(debt - Line, RAY));
+            }
```
Comment (iamchrissmith) [inline src/DssDirectDepositHub.sol:320]: how would you feel about moving this to an internal function to encapsulate this specific logic?

**Review by hexonaut — APPROVED** [review]
This is much cleaner, I like it.

File: src/DssDirectDepositHub.sol [hunk 2]
```diff
@@ -298,39 +305,41 @@ contract DssDirectDepositHub {
             );
         } else {
             // Normal path
+
+            // Determine if it needs to unwind due debt ceilings
```
Comment (talbaneth) [inline src/DssDirectDepositHub.sol:309]:
```suggestion
            // Determine if it needs to unwind due to debt ceilings
```

File: src/DssDirectDepositHub.sol [hunk 3]
```diff
@@ -298,39 +305,41 @@ contract DssDirectDepositHub {
...
-                    );
-                } else {
-                    uint256 amount = targetAssets - currentAssets;
-                    if (_add(Art, amount) > lineWad) { // we do not have enough room in the debt ceiling to fully wind
-                        amount = lineWad - Art;
-                    }
-                    _wind(ilk_, pool, amount);
-                }
-            } else if (targetAssets < currentAssets) {
+            if (toUnwind > 0) { // Our debt is greater than our debt ceiling, we need to unwind
```
Comment (talbaneth) [inline src/DssDirectDepositHub.sol:328]:
```suggestion
            if (toUnwind > 0) {
```

File: src/DssDirectDepositHub.t.sol [hunk 4]
```diff
@@ -370,13 +370,13 @@ contract DssDirectDepositHubTest is DSTest {
...
-        d3mTestPlan.file("targetAssets", 55 * WAD);
-        vat.file("Line", 45 * RAD);
+        d3mTestPlan.file("targetAssets", 55 * WAD); // Increasing target in 5 WAD
+        vat.file("Line", vat.debt() - 5 * RAD);
         directDepositHub.exec(ilk);
 
         // Ensure we unwound our position to debt ceiling
         (uint256 ink, uint256 art) = vat.urns(ilk, address(d3mTestPool));
-        assertEq(ink, 45 * WAD);
+        assertEq(ink, 45 * WAD); // Instead of 5 WAD more results in 5 WAD less due debt ceiling
```
Comment (talbaneth) [inline src/DssDirectDepositHub.t.sol:379]: Should we add 2 tests similar to the above for which a `wind` operation ends up being ceiled by line and Line?

**Review by talbaneth — COMMENTED** [review]
Changes are good.
See small comment fixes and suggestion for 2 tests.

File: src/DssDirectDepositHub.sol (hunk 1 above)
Comment (iamchrissmith) [reply src/DssDirectDepositHub.sol]: i'm fine with leaving as-is

File: src/DssDirectDepositHub.t.sol (hunk 4 above)
Comment (gbalabasquer) [reply src/DssDirectDepositHub.t.sol:403]: @talbaneth tests added.

**Review by talbaneth — APPROVED** [review]
LGTM

**Pull request merged**